utils/errors.go

```diff
@@ -34,43 +25,11 @@ func (e ProcessingError) Cause() error {
 	return e.OriginalError
 }
 
-// ProcessingErrorFromErrorWithMessage converts an error into a ProcessingError
-func ProcessingErrorFromErrorWithMessage(
-	err error,
-	message string,
-	kafkaMessage kafka.Message,
-	logger *zerolog.Logger,
-) *ProcessingError {
-	temporary, backoff := ErrorIsTemporary(err, logger)
+// ProcessingErrorFromError - given an error turn it into a processing error
+func ProcessingErrorFromError(err error, temporary bool) *ProcessingError {
```

Comment:
How about this for the function signature instead? 
```go
func NewProcessingError(cause error, isTemporary bool) error
``

Reply:
Fair enough. This and its implications are done.